ADR/ADR-Guidelines.md

```diff
@@ -0,0 +1,140 @@
+# ECMWF Architectural Decision Records (ADR) Guidelines
+
+This document provides a Guidelines and a template for creating Architectural Decision Records (ADRs). Each ADR should be a standalone document that captures a specific architectural decision, its context, and its consequences.
+
+## Template for Architectural Decision Records (ADRs)
+
+Please find the template for creatinsg ADRs in the [ADR-Template.md](./ADR-Template.md) file. This template should be used for all ADRs to ensure consistency and clarity.
+
+## Purpose
+
+Architectural Decision Records (ADRs) are lightweight documents that capture important architectural decisions made during the development of ECMWF systems. They provide historical context, rationale, and consequences of decisions to help current and future team members understand the evolution of our architecture.
+
+ADRs are living documents that serve the team. We should focus on creating useful, accessible records that will help current and future team members understand and build upon the architectural decisions that shape ECMWF's systems.
+
+## When to Write an ADR
+
+Create an ADR when making decisions that:
+
+- **Significantly impact system architecture** - Architectural choices, framework selections, major design patterns
+- **Affect multiple teams or systems** - Cross-cutting concerns, shared libraries, integration patterns
+- **Have long-term consequences** - Technology stack decisions, data models, security approaches, scalability strategies
+- **Require justification and analysis** - Trade-offs between options, risk assessments, cost-benefit
+- **Involve trade-offs between alternatives** - For example performance vs. maintainability, cost vs. functionality
+- **Establish important conventions** - Coding standards, deployment processes, monitoring strategies
+- **Address compliance or regulatory requirements** - Adherance to standards like OGC and Data governance, but also security policies and identity managment
```

Comment:
Given the broadness of this list, ADR is a slight misnomer as it imo applies only to part -- conventions and governance aren't really "architecture". Can't come up with much better name, Major Decision Record is perhaps too generic. The reason why I voice this is that when this is communicated, it should be emphasized it's not *only* for pure techno archy stuff

Reply:
Although I agree with your point, and it does go beyond architectural decisions, I would prefer to keep ADR as it is a common name in the industry and is useful as a communication tool. Maybe we can think of it as the 'Architecture' of the development process too ;-) 
Hopefully we will be focusing more on true architecture decisions that process decisions.